go.mod

```diff
@@ -81,7 +81,7 @@ require (
 	k8s.io/apimachinery v0.32.3
 	k8s.io/client-go v0.32.3
 	k8s.io/component-base v0.32.3
-	k8s.io/cri-api v0.32.3
+	k8s.io/cri-api v0.33.0-beta.0.0.20250313010358-ab383b81657e
```

Comment:
Would it be negotiable with cri-api maintainers to use Go 1.23?

Reply:
@samuelkarp recently opened a disucssion in kubernetes https://github.com/kubernetes/kubernetes/issues/130799

For `cri-api` specifically, my understanding is, as it's a staging module of k8s, changing it requires changing k8s' Go version (which is unlikely to revert back to 1.23 IMO).

From the discussion it seems 2 proposals:

1. move `cri-api` to a non-staging module (https://github.com/kubernetes/kubernetes/issues/130799#issuecomment-2722479805), which may take long I think.
2. copy the cri-api `api.proto` and generate go code ourselves (https://github.com/kubernetes/kubernetes/issues/130799#issuecomment-2724160839).

Reply:
> For cri-api specifically, my understanding is, as it's a staging module of k8s, changing it requires changing k8s' Go version (which is unlikely to revert back to 1.23 IMO).

It is a staging module and the go version is same across the k/k repo.

I think we might have to follow option 2 for our immediate use case.

Reply:
discussed briefly with derek and sasha .. we should just bring the proto over and build it ourselves... 2)

Reply:
Now that the proto is copied over, are these bumps to cri-api and Go still necessary?

Reply:
> Now that the proto is copied over,

I don't think that's been done :) (one relevant effort is https://github.com/containerd/containerd/pull/11695)